### PR TITLE
feat: basic TODO app; no bells, no whistles

### PR DESCRIPTION
**Pull request by jdevries3133**
I was wishing this was present when trying to learn from the examples in the repo – just a basic client-side TODO app without any server-side code or isomorphism.

I feel like this specific implementation probably stinks – I'm totally new to both Leptos and Rust, so happy to take any feedback & no hard feelings if there isn't interest in pulling an example like this into the repo. I fully appreciate my naive perspective here :)

### CODE REVIEW

**Comment by phillipbaird** [comment]
Would Vec.retain() be a simpler way to implement delete_todo?

**Comment by jdevries3133** [comment]
>  Would Vec.retain() be a simpler way to implement delete_todo?

Yeah, that's a definite improvement, thank you!

**Comment by jdevries3133** [comment]
Maybe this could be called `todo_component`? I think that not everyone is familiar with the MVC pattern, and this is line-for-line akin to a todo list app that you might see taught to React developers. Nonetheless, my feelings won't be hurt if you think this doesn't fit! I think you are right that the same ideas are represented already in other examples.

File: examples/todo_app_basic/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+#[derive(Eq, Hash, PartialEq, Clone, Debug)]
+struct TodoItem {
+    id: u32,
+    name: String,
+}
+
+#[derive(Copy, Clone)]
+struct TodoGenerator {
```
Comment (flosse) [inline examples/todo_app_basic/src/main.rs:14]: This should be a `struct TodoId` that has a `fn next(&self)` or so but you don't want to have a `TodoGenerator`.

**Comment by gbj** [comment]
@jdevries Sorry for my slow response here. Thanks for your contribution. We have created a new `projects` directory to house a wider variety of projects without the same maintenance burden as examples — Would you be interested in updating your PR and adding it to `projects` instead? I totally understand if not, it's been a while :-)

**Comment by gbj** [comment]
@jdevries Sorry for my slow response here. Thanks for your contribution. We have created a new `projects` directory to house a wider variety of projects without the same maintenance burden as examples — Would you be interested in updating your PR and adding it to `projects` instead? I totally understand if not, it's been a while :-)

**Pull request closed without merging**